### DISCUSSION TITLE
Pull Request Closed

### DISCUSSION THREAD

**Discussion by sadiefp** [nektos/act#2548] [Q&A]
Is there a way to specify a pull request closed event? 

I am sorry if this is documented, but I looked through discussions and documentation and I'm unsure if this is possible

**Comment by wellkilo** [comment]
Yes. The event name remains pull_request; the “closed” part comes from the event payload's action field.

Create an event.json containing at least the fields your workflow/actions read, for example:

    {
      "action": "closed",
      "number": 42,
      "pull_request": {
        "number": 42,
        "merged": true,
        "base": { "ref": "main" },
        "head": { "ref": "feature", "sha": "YOUR_HEAD_SHA" }
      },
      "repository": {
        "default_branch": "main",
        "full_name": "owner/repo"
      }
    }

Then run:

    act pull_request -e event.json

For a closed-but-not-merged PR set pull_request.merged to false. This allows conditions such as these to behave correctly:

    on:
      pull_request:
        types: [closed]

    if: github.event.pull_request.merged == true

-e is the short form of --eventpath (“path to event JSON file”) in act's [CLI definition](https://github.com/nektos/act/blob/4f411281417e88660bea1c1a1749aa71ae0bd60f/cmd/root.go#L88). act copies that payload to GITHUB_EVENT_PATH and uses the positional pull_request argument for workflow planning.

If an action reads more of the GitHub webhook schema, use a real sanitized pull_request.closed webhook payload rather than the minimal example. You can save one from GitHub's webhook delivery view or from gh api repos/OWNER/REPO/pulls/42 and wrap/add the action and repository fields.
